### DISCUSSION TITLE
Rerendering of wiki links

### DISCUSSION THREAD

**Discussion by Morgul** [requarks/wiki#5259] [Help / Questions]
I'm not sure if my usage of wiki.js is just different from normal, or if I've just missed something obvious. I'm currently using wiki.js as a collaborative tool for world building. It's a smallish wiki right now, with only about a hundred pages. However, we tend to write pages with a lot of cross references (i.e. links to other pages) and we don't necessarily build them in order. For example, if we have Page A that references Page B, it doesn't mean Page B exists yet; in fact most of the time it won't. We know what the link _will_ be, because we've agreed on a method for organizing the pages. (For example, all important locations will live under `/locations/some-location`, so when we decide we need a new location while working on a different page, we just drop a link to `/locations/new-location` and then come back and write that page.)

What this leads to is a lot of pages being created after they've been referenced, so the pages that link to them still render those links as missing. A global rerender fixes everything, but at 100 pages, it takes minutes to rerender everything, and I'm afraid that once this reaches several hundred pages that will become untenable.

Is there a way to just rerender a single page? (Other than a trivial edit of adding/removing a newline.) Or to make wiki.js pick up on the fact that I've just filled in a page, so now it can rerender all the pages that link to this one? Or, baring anything else, is there any simple way to run a scheduled job to just rerender the whole wiki every night?

**Comment by sb-ubnity** [comment]
Same question here :+1: 

Just FYI: the workaround is as you mentioned, but you don't actually need to add or remove anything. You can just go to edit mode and immediately save without making any changes to the page, it will cause re-render.

However, as an (expected) side-effect, it also creates an entry in the page version history, with an empty diff to the previous version.

Btw, another situation when you may want or need re-render is when you change the rendering settings (though probably in that case most of the time you would want to re-render all pages rather than just one).

**Comment by WTFSajuuk** [comment]
This is an extremely old post but I ran into the same issue. So I will leave a answer here if someone stumbles here somehow. 
I am updating my pages using a python script to keep documentation and code in sync and every time I was done I had to re render the entire site. The solution I found(I am using version 2.5.304) is to, after updating the page, use the graphql api. After updating the page just call https://your-host/graphql with the query below. That will re-render only the required page with the id $page_id

```
mutation { 
    pages {
        render(id: $page_id)
        {responseResult{succeeded errorCode slug message __typename} __typename} 
        __typename
        }
}
```

**Comment by Sergeus** [comment]
It looks like this is tracked as a currently open bug, #1614
